From 7dc378ccfba7f5be9d28b55b9ac8cdec8e4a2e08 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 7 Oct 2016 14:56:10 -0500
Subject: [PATCH 5/5] HACK: am335x-bone-common.dtsi: pwr button

new
/dev/input/event0:	tps65217_pwr_but
Select the device event number [0-0]: 0
Input driver version is 1.0.1
Input device ID: bus 0x18 vendor 0x0 product 0x0 version 0x0
Input device name: "tps65217_pwr_but"
Supported events:
  Event type 0 (EV_SYN)
  Event type 1 (EV_KEY)
    Event code 116 (KEY_POWER)
Properties:
Testing ... (interrupt to exit)

old
/dev/input/event0:	tps65217_pwr_but
Select the device event number [0-0]: 0
Input driver version is 1.0.1
Input device ID: bus 0x0 vendor 0x0 product 0x0 version 0x0
Input device name: "tps65217_pwr_but"
Supported events:
  Event type 0 (EV_SYN)
  Event type 1 (EV_KEY)
    Event code 116 (KEY_POWER)
Properties:
Testing ... (interrupt to exit)
Event: time 1475870099.434203, type 1 (EV_KEY), code 116 (KEY_POWER), value 1
Event: time 1475870099.434203, -------------- EV_SYN ------------
Event: time 1475870099.660284, type 1 (EV_KEY), code 116 (KEY_POWER), value 0
Event: time 1475870099.660284, -------------- EV_SYN ------------

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/am335x-bone-common-no-capemgr.dtsi | 9 +++++++++
 arch/arm/boot/dts/am335x-bone-common.dtsi            | 9 +++++++++
 drivers/input/misc/tps65218-pwrbutton.c              | 2 +-
 3 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/am335x-bone-common-no-capemgr.dtsi b/arch/arm/boot/dts/am335x-bone-common-no-capemgr.dtsi
index d35961b..9dd36a6 100644
--- a/arch/arm/boot/dts/am335x-bone-common-no-capemgr.dtsi
+++ b/arch/arm/boot/dts/am335x-bone-common-no-capemgr.dtsi
@@ -239,6 +239,11 @@
 /include/ "tps65217.dtsi"
 
 &tps {
+	//interrupts = <GIC_SPI 7 IRQ_TYPE_NONE>; /* NMIn */
+	interrupts = <0 7 0>; /* NMIn */
+	interrupt-controller;
+	#interrupt-cells = <2>;
+
 	/*
 	 * Configure pmic to enter OFF-state instead of SLEEP-state ("RTC-only
 	 * mode") at poweroff.  Most BeagleBone versions do not support RTC-only
@@ -302,6 +307,10 @@
 			regulator-always-on;
 		};
 	};
+
+	tps65217-pwrbutton {
+		compatible = "ti,tps65217-pwrbutton";
+	};
 };
 
 &cpsw_emac0 {
diff --git a/arch/arm/boot/dts/am335x-bone-common.dtsi b/arch/arm/boot/dts/am335x-bone-common.dtsi
index cdda427..635a6e3 100644
--- a/arch/arm/boot/dts/am335x-bone-common.dtsi
+++ b/arch/arm/boot/dts/am335x-bone-common.dtsi
@@ -288,6 +288,11 @@
 /include/ "tps65217.dtsi"
 
 &tps {
+	//interrupts = <GIC_SPI 7 IRQ_TYPE_NONE>; /* NMIn */
+	interrupts = <0 7 0>; /* NMIn */
+	interrupt-controller;
+	#interrupt-cells = <2>;
+
 	/*
 	 * Configure pmic to enter OFF-state instead of SLEEP-state ("RTC-only
 	 * mode") at poweroff.  Most BeagleBone versions do not support RTC-only
@@ -351,6 +356,10 @@
 			regulator-always-on;
 		};
 	};
+
+	tps65217-pwrbutton {
+		compatible = "ti,tps65217-pwrbutton";
+	};
 };
 
 &cpsw_emac0 {
diff --git a/drivers/input/misc/tps65218-pwrbutton.c b/drivers/input/misc/tps65218-pwrbutton.c
index 3273217..6f26022 100644
--- a/drivers/input/misc/tps65218-pwrbutton.c
+++ b/drivers/input/misc/tps65218-pwrbutton.c
@@ -36,7 +36,7 @@ struct tps6521x_data {
 static const struct tps6521x_data tps65217_data = {
 	.reg_status = TPS65217_REG_STATUS,
 	.pb_mask = TPS65217_STATUS_PB,
-	.name = "tps65217_pwrbutton",
+	.name = "tps65217_pwr_but",
 };
 
 static const struct tps6521x_data tps65218_data = {
-- 
2.9.3

